File: task_equivalents/llm_translations/gpt/optimized/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.formula.api as smf

auto = pd.read_csv("data/csv/Auto.csv", na_values="?").dropna()

model1 = smf.ols("mpg ~ horsepower", data=auto).fit()
print(model1.summary())

new_data = pd.DataFrame({"horsepower": [98]})

conf_pred = model1.get_prediction(new_data).summary_frame(alpha=0.05)
print(conf_pred[["mean", "mean_ci_lower", "mean_ci_upper"]])

pred_pred = model1.get_prediction(new_data).summary_frame(alpha=0.05)
print(pred_pred[["mean", "obs_ci_lower", "obs_ci_upper"]])

File: task_equivalents/llm_translations/gpt/optimized/r/statistical_modelling_analysis/01_linear_regression_inference.R
library(readr)

Auto <- read_csv("data/csv/Auto.csv", na = "?", show_col_types = FALSE) |>
tidyr::drop_na()

x <- Auto$horsepower
y <- Auto$mpg

model1_fit <- lm(y ~ x)
print(summary(model1_fit))

xnew <- data.frame(x = 98)
pred_conf <- predict(model1_fit, newdata = xnew, interval = "confidence", level = 0.95)
pred_pred <- predict(model1_fit, newdata = xnew, interval = "prediction", level = 0.95)

print(as.data.frame(pred_conf)[, c("fit", "lwr", "upr"), drop = FALSE])
print(as.data.frame(pred_pred)[, c("fit", "lwr", "upr"), drop = FALSE])